### PR TITLE
test(core-jest-matchers): increase coverage

### PR DESCRIPTION
**Pull request by JeremiGendron**
## Proposed changes

#1967

I've added tests following the existing implementations for the api/ source files.

I've also added or changed existing tests to also check that upon failure the message is as defined in the source files.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (improve a current implementation without adding a new feature or fixing a bug)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Build (changes that affect the build system)
- [ ] Docs (documentation only changes)
- [x] Test (adding missing tests or fixing existing tests)
- [ ] Other... Please describe:

## Checklist

- [x] I have read the [CONTRIBUTING](https://docs.ark.io/guidebook/contribution-guidelines/contributing.html) documentation
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by JeremiGendron** [comment]
I may have included commits from my previous branch

File: packages/core-jest-matchers/__tests__/transactions/valid.test.ts [hunk 1]
```diff
@@ -24,6 +24,6 @@ describe(".toBeValidTransaction", () => {
 
     test("fails when given an invalid transaction", () => {
         transaction.fee = "invalid" as any;
-        expect(transaction).not.toBeValidTransaction();
+        expect(expect(transaction).toBeValidTransaction).toThrowError("Expected value to be a valid transaction");
```
Comment (faustbrian) [inline packages/core-jest-matchers/__tests__/transactions/valid.test.ts]: What exactly is this supposed to do?

File: packages/core-jest-matchers/__tests__/transactions/types/vote.test.ts [hunk 2]
```diff
@@ -9,6 +9,6 @@ describe(".toBeVoteType", () => {
     });
 
     test("fails when given an invalid transaction", () => {
-        expect({ type: "invalid" }).not.toBeVoteType();
+        expect(expect({ type: "invalid" }).toBeVoteType).toThrowError("Expected value to be a valid VOTE transaction.");
```
Comment (faustbrian) [inline packages/core-jest-matchers/__tests__/transactions/types/vote.test.ts]: What exactly is this supposed to do?

**Comment by JeremiGendron** [comment]
They're being called by jest, there's no other way of verifying the returned message. If I pass the invocation it'll return a failed test and if I pass the function by reference it'll evaluate it as a sandboxed matcher.

Basically it was needed to verify the result of a failing test without actually creating a failing test, it seems jest is able to evaluate the test without running it with this method.

**Comment by JeremiGendron** [comment]
I played around with it a bit, if it's preferred I'll add back the existing tests for usability references and keep these in for coverage (unless otherwise noted).

**Pull request merged**